### PR TITLE
zphysics: Added a few missing API elements.

### PR DESCRIPTION
**Pull request by Srekel**
I'll likely add more later on, but wanted to get this pushed. 

Added functions:
**JPC_BodyInterface_GetPosition**
**JPC_BodyInterface_GetRotation**

Also added the **JPC_PhysicsUpdateError** return value to **JPC_PhysicsSystem_Update**.

### CODE REVIEW

File: libs/zphysics/libs/JoltC/JoltPhysicsC.h [hunk 1]
```diff
@@ -59,6 +59,15 @@ enum
     JPC_MAX_PHYSICS_BARRIERS = 8
 };
 
+typedef uint8_t JPC_PhysicsUpdateError;
```
Comment (michal-z) [inline libs/zphysics/libs/JoltC/JoltPhysicsC.h:62]:
```suggestion
typedef uint8_t JPC_PhysicsUpdateResult;
```

File: libs/zphysics/libs/JoltC/JoltPhysicsC.h [hunk 2]
```diff
@@ -59,6 +59,15 @@ enum
...
+enum
+{
+    JPC_PHYSICS_UPDATE_ERROR_NONE               = 0,
```
Comment (michal-z) [inline libs/zphysics/libs/JoltC/JoltPhysicsC.h:65]:
```suggestion
    JPC_PHYSICS_UPDATE_SUCCESS               = 0,
```

File: libs/zphysics/libs/JoltC/JoltPhysicsC.h [hunk 3]
```diff
@@ -1166,10 +1175,18 @@ JPC_BodyInterface_GetPointVelocity(const JPC_BodyInterface *in_iface,
...
+                              JPC_BodyID in_body_id,
+                              JPC_Real out_position[3]);
+JPC_API void
 JPC_BodyInterface_GetCenterOfMassPosition(const JPC_BodyInterface *in_iface,
                                           JPC_BodyID in_body_id,
                                           JPC_Real out_position[3]);
 JPC_API void
+JPC_BodyInterface_GetRotation(const JPC_BodyInterface *in_iface,
+                              JPC_BodyID in_body_id,
+                              JPC_Real out_rotation[4]);
```
Comment (michal-z) [inline libs/zphysics/libs/JoltC/JoltPhysicsC.h:1188]: Rotation always uses floats.
```suggestion
                              float out_rotation[4]);
```

**Review by michal-z — CHANGES_REQUESTED** [review]
Thanks!

Few minor things.
* Rotations always use single precision so we need to use `float` instead of `JPC_Real`
* Let's rename JPC_PhysicsUpdateError -> JPC_PhysicsUpdateResult

File: libs/zphysics/libs/JoltC/JoltPhysicsC.h (hunk 1 above)
Comment (Srekel) [reply libs/zphysics/libs/JoltC/JoltPhysicsC.h:62]: This is fine with me (that's what I would have named it too :) ), but just wanted to confirm given that the original enum is called Error:

external\zig-gamedev\libs\zphysics\libs\Jolt\Physics\EPhysicsUpdateError.h

/// Enum used by PhysicsSystem to report error conditions during the PhysicsSystem::Update call. This is a bit field, multiple errors can trigger in the same update.
enum class EPhysicsUpdateError : uint32
{
	None					= 0,			///< No errors
	ManifoldCacheFull		= 1 << 0,		///< The manifold cache is full, this means that the total number of contacts between bodies is too high. Some contacts were ignored. Increase inMaxContactConstraints in PhysicsSystem::Init.
	BodyPairCacheFull		= 1 << 1,		///< The body pair cache is full, this means that too many bodies contacted. Some contacts were ignored. Increase inMaxBodyPairs in PhysicsSystem::Init.
	ContactConstraintsFull	= 1 << 2,		///< The contact constraints buffer is full. Some contacts were ignored. Increase inMaxContactConstraints in PhysicsSystem::Init.
};

**Comment by michal-z** [comment]
@Srekel I did minor tweaks to the PR.

**Pull request merged**